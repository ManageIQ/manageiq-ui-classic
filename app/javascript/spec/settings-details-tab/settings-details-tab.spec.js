import React from 'react';
import fetchMock from 'fetch-mock';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import settingsDetailsTab from '../../components/settings-details-tab';

describe('SettingsDetailsTab Component', () => {
  const region = {
    id: 1,
    region: 0,
    description: 'Region 3',
    maintenance_zone_id: 1,
  };

  const scanItemsCount = 3;
  const zonesCount = 1;
  const miqSchedulesCount = 0;

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  it('should render settings details tab', async() => {
    const wrapper = shallow(<settingsDetailsTab
      region={region}
      scanItemsCount={scanItemsCount}
      zonesCount={zonesCount}
      miqSchedulesCount={miqSchedulesCount}
    />);

    await new Promise((resolve) => {
      setImmediate(() => {
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
        resolve();
      });
    });
  });
});
