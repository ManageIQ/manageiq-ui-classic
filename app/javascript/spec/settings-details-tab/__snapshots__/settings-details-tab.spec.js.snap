// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsDetailsTab Component should render settings details tab 1`] = `
<settingsDetailsTab
  miqSchedulesCount={0}
  region={
    Object {
      "description": "Region 3",
      "id": 1,
      "maintenance_zone_id": 1,
      "region": 0,
    }
  }
  scanItemsCount={3}
  zonesCount={1}
/>
`;
